In [3]:
# List of RTD function calls for AAPL
rtd_calls_list = [
    '=RTD(""tos.rtd"", , ""MARK"", ""AAPL"")',
    '=RTD("tos.rtd", , "LAST", "AAPL")',
    '=RTD("tos.rtd", , "LAST_SIZE", "AAPL")',
    '=RTD("tos.rtd", , "BID_SIZE", "AAPL")',
    '=RTD("tos.rtd", , "BID", "AAPL")',
    '=RTD("tos.rtd", , "ASK", "AAPL")',
    '=RTD("tos.rtd", , "ASK_SIZE", "AAPL")',
    '=RTD("tos.rtd", , "NET_CHANGE", "AAPL")',
    '=RTD("tos.rtd", , "PERCENT_CHANGE", "AAPL")',
    '=RTD("tos.rtd", , "MARK_PERCENT_CHANGE", "AAPL")',
    '=RTD("tos.rtd", , "PUT_CALL_RATIO", "AAPL")',
    '=RTD("tos.rtd", , "CALL_VOLUME_INDEX", "AAPL")',
    '=RTD("tos.rtd", , "PUT_VOLUME_INDEX", "AAPL")',
    '=RTD("tos.rtd", , "OPTION_VOLUME_INDEX", "AAPL")',
    '=RTD("tos.rtd", , "BACK_EX_MOVE", "AAPL")',
    '=RTD("tos.rtd", , "FRONT_EX_MOVE", "AAPL")',
    '=RTD("tos.rtd", , "EX_MOVE_DIFF", "AAPL")',
    '=RTD("tos.rtd", , "IMPL_VOL", "AAPL")',
    '=RTD("tos.rtd", , "Spreads", "AAPL")',
    '=RTD("tos.rtd", , "MRKT_MKR_MOVE", "AAPL")',
    '=RTD("tos.rtd", , "MarketSentiment", "AAPL")',
    '=RTD("tos.rtd", , "VOLUME", "AAPL")'
]

# Each RTD call is already a string, so we can just print them out or use them as needed
rtd_calls_list


['=RTD(""tos.rtd"", , ""MARK"", ""AAPL"")',
 '=RTD("tos.rtd", , "LAST", "AAPL")',
 '=RTD("tos.rtd", , "LAST_SIZE", "AAPL")',
 '=RTD("tos.rtd", , "BID_SIZE", "AAPL")',
 '=RTD("tos.rtd", , "BID", "AAPL")',
 '=RTD("tos.rtd", , "ASK", "AAPL")',
 '=RTD("tos.rtd", , "ASK_SIZE", "AAPL")',
 '=RTD("tos.rtd", , "NET_CHANGE", "AAPL")',
 '=RTD("tos.rtd", , "PERCENT_CHANGE", "AAPL")',
 '=RTD("tos.rtd", , "MARK_PERCENT_CHANGE", "AAPL")',
 '=RTD("tos.rtd", , "PUT_CALL_RATIO", "AAPL")',
 '=RTD("tos.rtd", , "CALL_VOLUME_INDEX", "AAPL")',
 '=RTD("tos.rtd", , "PUT_VOLUME_INDEX", "AAPL")',
 '=RTD("tos.rtd", , "OPTION_VOLUME_INDEX", "AAPL")',
 '=RTD("tos.rtd", , "BACK_EX_MOVE", "AAPL")',
 '=RTD("tos.rtd", , "FRONT_EX_MOVE", "AAPL")',
 '=RTD("tos.rtd", , "EX_MOVE_DIFF", "AAPL")',
 '=RTD("tos.rtd", , "IMPL_VOL", "AAPL")',
 '=RTD("tos.rtd", , "Spreads", "AAPL")',
 '=RTD("tos.rtd", , "MRKT_MKR_MOVE", "AAPL")',
 '=RTD("tos.rtd", , "MarketSentiment", "AAPL")',
 '=RTD("tos.rtd", , "VOLUME", "AAPL")']

In [4]:
new_list = [
    '=RTD(""tos.rtd"", , ""MARK"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""LAST"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""LAST_SIZE"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""BID_SIZE"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""BID"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""ASK"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""ASK_SIZE"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""NET_CHANGE"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""PERCENT_CHANGE"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""MARK_PERCENT_CHANGE"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""PUT_CALL_RATIO"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""CALL_VOLUME_INDEX"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""PUT_VOLUME_INDEX"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""OPTION_VOLUME_INDEX"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""BACK_EX_MOVE"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""FRONT_EX_MOVE"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""EX_MOVE_DIFF"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""IMPL_VOL"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""Spreads"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""MRKT_MKR_MOVE"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""MarketSentiment"", ""AAPL"")',
    '=RTD(""tos.rtd"", , ""VOLUME"", ""AAPL"")'
]

new_list

['=RTD(""tos.rtd"", , ""MARK"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""LAST"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""LAST_SIZE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""BID_SIZE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""BID"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""ASK"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""ASK_SIZE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""NET_CHANGE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""PERCENT_CHANGE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""MARK_PERCENT_CHANGE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""PUT_CALL_RATIO"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""CALL_VOLUME_INDEX"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""PUT_VOLUME_INDEX"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""OPTION_VOLUME_INDEX"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""BACK_EX_MOVE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""FRONT_EX_MOVE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""EX_MOVE_DIFF"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""IMPL_VOL"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""Spreads"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""MRKT_MKR_MOVE"",

In [5]:
# You can add any ticker you want here. We will pull in the list of NQ+ES MAKE THIS FRONT END
fake_spy = ["AAPL", "AMZN", "QQQ"]


# Iterate through the calls. 
# AAPL needs to become 'x'
# We change AAPL in each row with 'x'
# We have a list of stocks that we want to update 'x' to be
# First, x = AAPL, then x = AMZN, then x = QQQ, for all tickers in 'fake_spy'

rtd_call_per_ticker = []

for Ticker in fake_spy:
    fake_rtd_calls = [
        '=RTD("tos.rtd", , "MARK", "' + Ticker + '")',
        '=RTD("tos.rtd", , "LAST", "' + Ticker + '")',
        '=RTD("tos.rtd", , "LAST_SIZE", "' + Ticker + '")',
        '=RTD("tos.rtd", , "BID_SIZE", "' + Ticker + '")',
        '=RTD("tos.rtd", , "BID", "' + Ticker + '")',
        '=RTD("tos.rtd", , "ASK", "' + Ticker + '")',
        '=RTD("tos.rtd", , "ASK_SIZE", "' + Ticker + '")',
        '=RTD("tos.rtd", , "NET_CHANGE", "' + Ticker + '")',
        '=RTD("tos.rtd", , "PERCENT_CHANGE", "' + Ticker + '")',
        '=RTD("tos.rtd", , "MARK_PERCENT_CHANGE", "' + Ticker + '")',
        '=RTD("tos.rtd", , "PUT_CALL_RATIO", "' + Ticker + '")',
        '=RTD("tos.rtd", , "CALL_VOLUME_INDEX", "' + Ticker + '")',
        '=RTD("tos.rtd", , "PUT_VOLUME_INDEX", "' + Ticker + '")',
        '=RTD("tos.rtd", , "OPTION_VOLUME_INDEX", "' + Ticker + '")',
        '=RTD("tos.rtd", , "BACK_EX_MOVE", "' + Ticker + '")',
        '=RTD("tos.rtd", , "FRONT_EX_MOvE", "' + Ticker + '")',
        '=RTD("tos.rtd", , "EX_MOVE_DIFF", "' + Ticker + '")',
        '=RTD("tos.rtd", , "IMPL_VOL", "' + Ticker + '")',
        '=RTD("tos.rtd", , "Spreads", "' + Ticker + '")',
        '=RTD("tos.rtd", , "MRKT_MKR_MOVE", "' + Ticker + '")',
        '=RTD("tos.rtd", , "MarketSentiment", "' + Ticker + '")',
        '=RTD("tos.rtd", , "VOLUME", "' + Ticker + '")',
    ]
    rtd_call_per_ticker.append(fake_rtd_calls)

rtd_call_per_ticker

[['=RTD("tos.rtd", , "MARK", "AAPL")',
  '=RTD("tos.rtd", , "LAST", "AAPL")',
  '=RTD("tos.rtd", , "LAST_SIZE", "AAPL")',
  '=RTD("tos.rtd", , "BID_SIZE", "AAPL")',
  '=RTD("tos.rtd", , "BID", "AAPL")',
  '=RTD("tos.rtd", , "ASK", "AAPL")',
  '=RTD("tos.rtd", , "ASK_SIZE", "AAPL")',
  '=RTD("tos.rtd", , "NET_CHANGE", "AAPL")',
  '=RTD("tos.rtd", , "PERCENT_CHANGE", "AAPL")',
  '=RTD("tos.rtd", , "MARK_PERCENT_CHANGE", "AAPL")',
  '=RTD("tos.rtd", , "PUT_CALL_RATIO", "AAPL")',
  '=RTD("tos.rtd", , "CALL_VOLUME_INDEX", "AAPL")',
  '=RTD("tos.rtd", , "PUT_VOLUME_INDEX", "AAPL")',
  '=RTD("tos.rtd", , "OPTION_VOLUME_INDEX", "AAPL")',
  '=RTD("tos.rtd", , "BACK_EX_MOVE", "AAPL")',
  '=RTD("tos.rtd", , "FRONT_EX_MOvE", "AAPL")',
  '=RTD("tos.rtd", , "EX_MOVE_DIFF", "AAPL")',
  '=RTD("tos.rtd", , "IMPL_VOL", "AAPL")',
  '=RTD("tos.rtd", , "Spreads", "AAPL")',
  '=RTD("tos.rtd", , "MRKT_MKR_MOVE", "AAPL")',
  '=RTD("tos.rtd", , "MarketSentiment", "AAPL")',
  '=RTD("tos.rtd", , "VOLUME", "AAPL"

Now we are going to do the same with ALL stocks in both the NQ and ES

In [6]:
# Import the required libraries
import numpy as np
import pandas as pd

In [7]:
# Read csv, take a look
nq_df = pd.read_csv(("../algotrader2/resources/nq_stocks.csv"))
nq_df.head()

Symbol           Description      Category2  Category3  \
0   AAPL             Apple Inc  Common stocks  Large cap   
1   MSFT        Microsoft Corp  Common stocks  Large cap   
2   GOOG  Alphabet Inc Class C  Common stocks  Large cap   
3  GOOGL  Alphabet Inc Class A  Common stocks  Large cap   
4   AMZN       Amazon.Com Inc.  Common stocks  Large cap   

              GICS Sector          Market cap Dividend yield Country   Action  
0  Information Technology  $2,514,581,877,780          0.58%     USA  Analyze  
1  Information Technology  $2,066,846,488,973          0.93%     USA  Analyze  
2  Communication Services  $1,263,113,280,000          0.00%     USA  Analyze  
3  Communication Services  $1,263,113,280,000          0.00%     USA  Analyze  
4  Consumer Discretionary  $1,011,507,010,613          0.00%     USA  Analyze

In [8]:
# Do the same for the snp 500 symbols
snp_df = pd.read_csv(("../algotrader2/resources/snp_stocks.csv"))
snp_df.head()

Symbol     Security             GICS Sector               GICS Sub-Industry  \
0    MMM           3M             Industrials        Industrial Conglomerates   
1    AOS  A. O. Smith             Industrials               Building Products   
2    ABT       Abbott             Health Care           Health Care Equipment   
3   ABBV       AbbVie             Health Care                   Biotechnology   
4    ACN    Accenture  Information Technology  IT Consulting & Other Services   

     Headquarters Location  Date added      CIK      Founded  
0    Saint Paul, Minnesota  1957-03-04    66740         1902  
1     Milwaukee, Wisconsin  2017-07-26    91142         1916  
2  North Chicago, Illinois  1957-03-04     1800         1888  
3  North Chicago, Illinois  2012-12-31  1551152  2013 (1888)  
4          Dublin, Ireland  2011-07-06  1467373         1989

In [9]:
# # Save the symbols from each df
# nq_symbols_df = nq_df["Symbol"]
# snp_symbols_df = snp_df["Symbol"]

# Save symbols from each df and put them together
snp_nq_symbols_df = pd.concat([nq_df["Symbol"], snp_df["Symbol"]], axis=0)

# Drop duplicates
snp_nq_symbols_df.drop_duplicates(inplace=True)

# View
snp_nq_symbols_df

0       AAPL
1       MSFT
2       GOOG
3      GOOGL
4       AMZN
       ...  
498      YUM
499     ZBRA
500      ZBH
501     ZION
502      ZTS
Name: Symbol, Length: 521, dtype: object

As we can see, the index is wrong. Somewhere we have duplicate indices, since we end at 502 but apparently have 521. We may come back to this, but I don't think it matters.

In [10]:
# Convert symbols to a list so that we can feed it into our rtd_call function

nq_snp_symbols_list = snp_nq_symbols_df.tolist()

print(f"Number of unique stocks: {len(nq_snp_symbols_list)}")
nq_snp_symbols_list[:10]


Number of unique stocks: 521


['AAPL',
 'MSFT',
 'GOOG',
 'GOOGL',
 'AMZN',
 'NVDA',
 'TSLA',
 'META',
 'ASML',
 'AVGO']

In [11]:
fake_list = []


for Ticker in fake_spy:
    fake_calls = [
        '=RTD(""tos.rtd"", , ""MARK"", ""' + Ticker + '"")',
        '=RTD(""tos.rtd"", , ""LAST"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""LAST_SIZE"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""BID_SIZE"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""BID"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""ASK"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""ASK_SIZE"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""NET_CHANGE"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""PERCENT_CHANGE"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""MARK_PERCENT_CHANGE"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""PUT_CALL_RATIO"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""CALL_VOLUME_INDEX"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""PUT_VOLUME_INDEX"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""OPTION_VOLUME_INDEX"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""BACK_EX_MOVE"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""FRONT_EX_MOvE"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""EX_MOVE_DIFF"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""IMPL_VOL"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""Spreads"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""MRKT_MKR_MOVE"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""MarketSentiment"", ""Ticker"")',
        '=RTD(""tos.rtd"", , ""VOLUME"", ""Ticker"")'
    ]
    fake_list.append(fake_calls)
    

fake_list

[['=RTD(""tos.rtd"", , ""MARK"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""LAST"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""LAST_SIZE"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""BID_SIZE"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""BID"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""ASK"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""ASK_SIZE"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""NET_CHANGE"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""PERCENT_CHANGE"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""MARK_PERCENT_CHANGE"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""PUT_CALL_RATIO"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""CALL_VOLUME_INDEX"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""PUT_VOLUME_INDEX"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""OPTION_VOLUME_INDEX"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""BACK_EX_MOVE"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""FRONT_EX_MOvE"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""EX_MOVE_DIFF"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""IMPL_VOL"", ""Ticker"")',
  '=RTD(""tos.rtd"", , ""Spreads"",

In [22]:
# Skeleton list of lists. Each element is a list of all of calls we want to make, given each stock
rtd_call_per_ticker = []

# Iterate through the calls. 
# AAPL needs to become 'x'
# We change AAPL in each row with 'x'
# We have a list of stocks that we want to update 'x' to be
# First, x = AAPL, then x = AMZN, then x = QQQ, for all tickers in nq_snp_symbols_list


# Iterate through tickers in our combined nq/snp list.
# Keep adding those lists to our list of lists.
# Fixed the double quotation problem with help from ChatGPT
for Ticker in nq_snp_symbols_list:
    real_call_list = [
    '=RTD(""tos.rtd"", , ""MARK"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""LAST"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""LAST_SIZE"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""BID_SIZE"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""BID"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""ASK"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""ASK_SIZE"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""NET_CHANGE"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""PERCENT_CHANGE"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""MARK_PERCENT_CHANGE"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""PUT_CALL_RATIO"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""CALL_VOLUME_INDEX"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""PUT_VOLUME_INDEX"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""OPTION_VOLUME_INDEX"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""BACK_EX_MOVE"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""FRONT_EX_MOvE"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""EX_MOVE_DIFF"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""IMPL_VOL"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""Spreads"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""MRKT_MKR_MOVE"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""MarketSentiment"", ""' + Ticker + '"")',
    '=RTD(""tos.rtd"", , ""VOLUME"", ""' + Ticker + '"")',
    ]
    rtd_call_per_ticker.append(real_call_list)
    

In [23]:
rtd_call_per_ticker

[['=RTD(""tos.rtd"", , ""MARK"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""LAST"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""LAST_SIZE"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""BID_SIZE"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""BID"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""ASK"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""ASK_SIZE"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""NET_CHANGE"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""PERCENT_CHANGE"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""MARK_PERCENT_CHANGE"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""PUT_CALL_RATIO"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""CALL_VOLUME_INDEX"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""PUT_VOLUME_INDEX"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""OPTION_VOLUME_INDEX"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""BACK_EX_MOVE"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""FRONT_EX_MOvE"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""EX_MOVE_DIFF"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""IMPL_VOL"", ""AAPL"")',
  '=RTD(""tos.rtd"", , ""Spreads"", ""AAPL"")',
  '=RTD(""tos.rtd"", 

Good. Now we have a list of lists. Ideally we don't want a list of lists, we just want a big list.

In [24]:
# Thanks ChatGPT for showing how to flatten a list.
mass_rtd_list = [item for sublist in rtd_call_per_ticker for item in sublist]

mass_rtd_list

['=RTD(""tos.rtd"", , ""MARK"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""LAST"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""LAST_SIZE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""BID_SIZE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""BID"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""ASK"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""ASK_SIZE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""NET_CHANGE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""PERCENT_CHANGE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""MARK_PERCENT_CHANGE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""PUT_CALL_RATIO"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""CALL_VOLUME_INDEX"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""PUT_VOLUME_INDEX"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""OPTION_VOLUME_INDEX"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""BACK_EX_MOVE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""FRONT_EX_MOvE"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""EX_MOVE_DIFF"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""IMPL_VOL"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""Spreads"", ""AAPL"")',
 '=RTD(""tos.rtd"", , ""MRKT_MKR_MOVE"",

Bingo. Now we have to format it to match what we will actually paste into our program. (Found by Alex, will link to the 2019 repository. The author used Visual Basics to program this.)

In [16]:
# Format to replicate, with 'mass_rtd_list'
# XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""LAST"", ""QQQ"" )"
# XL_Cell_2.Formula = "=RTD(""tos.rtd"", , ""LAST"", ""AAPL"" )"
# XL_Cell_3.Formula = "=RTD(""tos.rtd"", , ""BID"", ""AAPL"" )"
# ' ...
# XL_Cell_LAST.Formula = "=RTD(""tos.rtd"", , ""INDICATOR"", ""TICKER"" )"
# XL_Cell_LAST.Formula = "=RTD(""tos.rtd"", , ""INDICATOR2"", ""TICKER2"" )"


# First thing to do is to fix the quotations. It gives ' ... we need  "
# DONE

In [25]:
# Now we add the XL_Cell_3 part.
# We are creating strings

string1 = "XL_Cell_1.Formula = " + "\""  + mass_rtd_list[1] + "\""

print(string1)

XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""LAST"", ""AAPL"")"


In [26]:
# Good. Now iterate and save into a final list.

final_rtd_calls = []
x = 0

for x in range(len(mass_rtd_list)):
    string1 = "XL_Cell_1.Formula = " + "\""  + mass_rtd_list[x] + "\""
    final_rtd_calls.append(string1)
    x+1



final_rtd_calls

['XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""MARK"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""LAST"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""LAST_SIZE"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""BID_SIZE"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""BID"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""ASK"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""ASK_SIZE"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""NET_CHANGE"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""PERCENT_CHANGE"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""MARK_PERCENT_CHANGE"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""PUT_CALL_RATIO"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""CALL_VOLUME_INDEX"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""PUT_VOLUME_INDEX"", ""AAPL"")"',
 'XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""OPTION_VOLUME_INDEX"

In [28]:
print(final_rtd_calls[10])
print(final_rtd_calls[50])
print(final_rtd_calls[100])

XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""PUT_CALL_RATIO"", ""AAPL"")"
XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""ASK_SIZE"", ""GOOG"")"
XL_Cell_1.Formula = "=RTD(""tos.rtd"", , ""PUT_VOLUME_INDEX"", ""AMZN"")"


Bingo. Now we can copy and paste the output into our program.